presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PartialAggregationPushDown.java

```diff
@@ -87,6 +87,14 @@ public PlanNode visitAggregation(AggregationNode node, RewriteContext<Void> cont
                 return context.defaultRewrite(node);
             }
 
+            boolean decomposable = node.getFunctions().values().stream()
```

Comment:
There was no reason. Code after and before my change behaves identically. Regardless whether this cut off was activated earlier/later or there was another check that has failed, all would call `context.defaultRewrite(node)`, so there is no preference in what order those "cut offs" are being checked.

I just need this `decomposable` value earlier in the later commit, to make a `checkState`.